crates/net/eth-wire-types/src/message.rs

```diff
@@ -57,7 +59,11 @@ impl<N: NetworkPrimitives> ProtocolMessage<N> {
         let message_type = EthMessageID::decode(buf)?;
 
         let message = match message_type {
-            EthMessageID::Status => EthMessage::Status(Status::decode(buf)?),
+            EthMessageID::Status => EthMessage::Status(if version < EthVersion::Eth69 {
+                StatusPayload::Legacy(Status::decode(buf)?)
+            } else {
+                StatusPayload::Eth69(StatusEth69::decode(buf)?)
```

Comment:
great, let's add a one liner why this is necessary and reference the eip for this:

https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7642.md